sourced/ml/core/utils/bblfsh.py

```diff
@@ -1,19 +1,66 @@
 from distutils.version import StrictVersion
+from typing import List, Sequence
 
-from bblfsh.client import BblfshClient
+from bblfsh.compat import CompatBblfshClient as BblfshClient
+from bblfsh.compat import CompatNodeIterator, Node
 
-BBLFSH_VERSION_LOW = "2.2"
-BBLFSH_VERSION_HIGH = "3.0"
+
+BBLFSH_VERSION_LOW = "3.0"
+BBLFSH_VERSION_HIGH = "4.0"
 
 
 def check_version(host: str = "0.0.0.0", port: str = "9432") -> bool:
     """
     Check if the bblfsh server version matches module requirements.
 
-    :param host: bblfsh server host
-    :param port: bblfsh server port
-    :return: True if bblfsh version specified matches requirements
+    :param host: bblfsh server host.
+    :param port: bblfsh server port.
+    :return: True if bblfsh version specified matches requirements.
     """
     # get version and remove leading 'v'
     version = StrictVersion(BblfshClient("%s:%s" % (host, port)).version().version.lstrip("v"))
     return StrictVersion(BBLFSH_VERSION_LOW) <= version < StrictVersion(BBLFSH_VERSION_HIGH)
+
+
+def iterate_children(nodes: Sequence[Node]) -> CompatNodeIterator:
+    """
+    Use this function to access the node.children attribute in a Babelfish v3 Node.
+
+    By default, node.children return the children of the nodes as Node objects or as dicts, but
+    Babelfish v2 returns only Node instances. This function adapts the returned bblfsh v3 children
+    to return only Node instances.
+
+    The node.children attribute is transformed to an iterator, and then filtered by a
+    CompatNodeIterator. The CompatNodeIterator transforms the dict nodes into Node instances.
+    """
+    def node_generator(node_seq):
+        for val in node_seq:
+            yield val
+    return CompatNodeIterator(node_generator(nodes))
+
+
+def get_node_tokens(node: Node) -> List[str]:
```

Comment:
This has a different signature than the original. I am not sure if `".".join(get_node_tokens(node))` would replicate the original one. (I don't know how it looked in the first place).

Unfortunatelly, the [`@token` field](https://docs.sourced.tech/babelfish/uast/semantic-uast#common-fields) describe here does not exists.